nixos/modules/services/security/step-ca.nix

```diff
@@ -0,0 +1,68 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.step-ca;
+  settingsFormat = (pkgs.formats.json { });
+in
+{
+  options = {
+    services.step-ca = {
+      enable = lib.mkEnableOption "the smallstep certificate authority server";
+      settings = lib.mkOption {
+        type = lib.types.anything;
+        description = "See https://smallstep.com/docs/step-ca/configuration. https://smallstep.com/docs/step-cli/basic-crypto-operations#run-an-offline-x509-certificate-authority may also be useful. The easiest way is probably to use the existing smallstep tools to set up the CA and then import the config using e.g. `builtins.fromJSON`";
+      };
+      intermediatePasswordFile = lib.mkOption {
+        type = lib.types.str;
+        default = null;
+        example = "/run/keys/smallstep-password";
+        description = "Path to the file containing the password for the intermediate certificate private key. WARNING: make sure to use a quoted absolute path or it is going to be copied to the nix store.";
+      };
+    };
+  };
+
+  config = lib.mkIf config.services.step-ca.enable (
+    let
+      configFile = settingsFormat.generate "ca.json" cfg.settings;
+    in
+    {
+      assertions =
+        [{
+          assertion = !lib.isStorePath cfg.intermediatePasswordFile;
+          message = "`services.step-ca.intermediatePasswordFile` points to a file in the nix store. You should use a quoted absolute path to prevent this.";
+        }];
+
+      systemd.packages = [ pkgs.step-ca ];
+
+      # configuration file indirection is needed to support reloading
+      environment.etc."smallstep-ca.json".source = configFile;
+
+      systemd.services."step-ca" = {
+        wantedBy = [ "multi-user.target" ];
+        reloadIfChanged = true;
```

Comment:
Should be unnecessary with `restartTriggers`.

Reply:
It depends if the intention is to restart the service every time configuration changes, as opposed to reloading the service.